sycl/doc/extensions/DeviceGlobal/SYCL_INTEL_device_global.asciidoc

```diff
@@ -0,0 +1,790 @@
+----
+
+[NOTE]
+====
+
+On a device, `device_global` has similar semantics to a reference wrapper.  The dot operator (`operator.`) cannot be overloaded, so a `get()` member is provided to allow a reference to be extracted directly when needed.  Some operators are declared in `device_global` that must be members (e.g. `operator[]` and `+operator->+`).  Note that other operators can be overloaded by specific `T` as free functions, which will be selected through implicit conversion to `T` in device functions.
+
+====
+
+
+The section below and the table following describe the constructors, member functions and factory methods for `device_global`.
+
+[source,c++]
+----
+namespace sycl::ext::oneapi {
+
+template <typename T, typename property_listT = property_list<>>
+class device_global {
+public:
+  using element_type = std::remove_extent_t<T>; 
+  
+  static_assert(std::is_trivially_default_constructible_v<T>,
+                "Type T must be trivially default constructable (until C++20 "
+                "consteval is supported and enabled)");
+
+  static_assert(std::is_trivially_destructible_v<T>,
+      "Type T must be trivially destructible.");
+
+  // Only default construction is allowed.  The underlying memory allocations
+  // of type T on devices will be zero-initialized before any non-initialization
+  // accesses occur.
+  device_global();
+
+  device_global(const device_global &) = delete;
+  device_global(const device_global &&) = delete;
+  device_global &operator=(const device_global &) = delete;
+  device_global &operator=(const device_global &&) = delete;
+
+  template <access::decorated IsDecorated>
+  multi_ptr<T, access::address_space::global_space, IsDecorated> get_multi_ptr() noexcept;
+  template <access::decorated IsDecorated>
+  multi_ptr<const T, access::address_space::global_space, IsDecorated> get_multi_ptr() const noexcept;
+
+  // Access the underlying data
+  operator T&() noexcept;
+  operator const T&() const noexcept;
+ 
+  T& get() noexcept;
+  const T& get() const noexcept;
+
+  // Enable assignments from underlying type
+  device_global& operator=(const T&) noexcept;
+
+  // Available if the operator[] is valid for objects of type T
+  using subscript_return_t = std::remove_reference_t<decltype(std::declval<T>()[std::ptrdiff_t{}])>;
+  subscript_return_t& operator[]( std::ptrdiff_t idx ) noexcept;
+  const subscript_return_t& operator[]( std::ptrdiff_t idx ) const noexcept;
+
+  // Available if the operator-> is valid for objects of type T
+  T& operator->() noexcept;
+  const T& operator->() const noexcept;
+
+  // Note that there is no need for "device_global" to define member functions for
+  // operators like "++", comparison, etc.  Instead, the type "T" need only define
+  // these operators as non-member functions.  Because there is an implicit conversion
+  // from "device_global" to "T&", the operations can be applied to objects of type
+  // "device_global<T>".
+
+  template<typename propertyT>
```

Comment:
Nit: Same comment as above: I recommend renaming `propertyT` to `PropertyT`.  There are other instances of `propertyT` in the document, but I won't highlight them all.

Reply:
Fixed.